exporter/cat1_view_helper.go

```diff
@@ -140,11 +177,31 @@ func valueOrDefault(val interface{}, def interface{}) interface{} {
 	return def
 }
 
-// conditional assignment. returns the second value only if the first value is zero
-// TODO: make arguments and return type interface{}. add "value is empty, zero, or nil" to description
-func condAssign(first int64, second int64) int64 {
-	if first != 0 {
-		return first
+// conditional assignment. returns the second value only if the first value is empty, zero, or nil
+func condAssign(first interface{}, second interface{}) interface{} {
+	switch val := first.(type) {
```

Comment:
I think there is a problem with this switch statement. You are assigning `val` the type of `first`, and then using `val` to test for zero values in the case bodies. I think what you want is:
```Go
switch first.(type) { 
    case string:
        if first != "" {
            return first
        }
    ...
```